.github/workflows/build_and_test.yaml

```diff
@@ -0,0 +1,33 @@
+name: Build and Test
```

Comment:
emm, actually this file need to be named `build_and_test.yml` for GitHub to pick it up.

Reply:
It should be fine as-is, it's running correctly on my branch (https://github.com/skriss/gateway/actions/runs/2392566235) as well as on a test PR I created in my fork (https://github.com/skriss/gateway/pull/1/checks). Most of our workflow files in Contour also use the `.yaml` extension.  I'm still not sure why it's not running on this PR, but maybe we can merge & see if it runs on `main` and on the next PR.

Reply:
OK let's give it a try as is.